build.gradle

```diff
@@ -14,47 +14,59 @@
  * limitations under the License.
  */
 
-apply plugin: 'groovy'
-apply plugin: 'java'
-apply plugin: 'com.github.ben-manes.versions'
-
-sourceCompatibility = JavaVersion.VERSION_1_7
-
-group = GROUP
-version = VERSION_NAME
-
 buildscript {
     repositories {
         maven { url "https://plugins.gradle.org/m2/" } // Mirrors jcenter() and mavenCentral()
     }
 
     dependencies {
-        // Check for plugin updates
-        classpath "com.github.ben-manes:gradle-versions-plugin:0.12.0"
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.1.1"
+        classpath "com.github.ben-manes:gradle-versions-plugin:0.14.0"
     }
 }
 
 repositories {
     maven { url "https://plugins.gradle.org/m2/" } // Mirrors jcenter() and mavenCentral()
 }
 
+apply plugin: "groovy"
+apply plugin: "kotlin"
+apply plugin: "java"
+apply plugin: "com.github.ben-manes.versions"
+
+sourceCompatibility = JavaVersion.VERSION_1_7
+
+group = GROUP
+version = VERSION_NAME
+
 jar {
     manifest {
         attributes(
-                'Implementation-Title': POM_ARTIFACT_ID,
-                'Implementation-Version': VERSION_NAME)
+                "Implementation-Title": POM_ARTIFACT_ID,
+                "Implementation-Version": VERSION_NAME)
     }
 }
 
+sourceSets {
+    main.java.srcDirs += "src/main/kotlin"
+}
+
 dependencies {
     compile gradleApi()
-    compile 'com.android.tools.build:gradle:2.3.0'
-    compile 'com.google.code.gson:gson:2.6.2'
+    compile "com.android.tools.build:gradle:2.3.0"
+    compile "com.google.code.gson:gson:2.8.0"
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jre7:1.1.0" // Make sure to use jre7
 
-    testCompile 'junit:junit:4.12'
-    testCompile('org.spockframework:spock-core:1.0-groovy-2.4') {
-        exclude module: 'groovy-all'
-    }
+    testCompile "junit:junit:4.12"
+    testCompile("org.spockframework:spock-core:1.0-groovy-2.4") { exclude module: "groovy-all" }
 }
 
-apply from: file('gradle/gradle-mvn-push.gradle')
+apply from: file("gradle/gradle-mvn-push.gradle")
+
+// Compile Kotlin then Groovy
+afterEvaluate {
+    compileKotlin.dependsOn = compileKotlin.taskDependencies.values - "compileJava"
```

Comment:
I had to add this to be able to EVEN start a slow 1:1 conversion. Gradle can't compile all 3 "together". You have to change the order.